docs/queries/regular-expressions.md

```diff
@@ -0,0 +1,130 @@
+- Regex searches are **case-sensitive**, unlike the simpler  `includes` and  `does not include`
+- A regex search can be made **insensitive** by appending a `i` flag after the closing `/`, for example: `/I aM cAsE INsensitive because of the LiTle i after the closing slash/i`
+- Tasks does not support multi-line regex searches, as each task is a single line.
+- Note that `tags` searches do not yet support regex searches.
+  - As a workaround, search `description` instead.
+
+## Special characters
+
+If using regex searches, it is important to be aware of the available special characters for several reasons:
+
+1. They enable complex queries to written in simple ways
+2. They can cause confusing results or broken searches, if not "escaped" in the search.
+
+Here are a few examples of the [many special characters](https://www.rexegg.com/regex-quickstart.html):
+
+- `.` matches any character
+- `^` matches the start of the string (but when `[^inside brackets]`, it means "not")
+- `$` matches the end of the string
+- `\` adds special meaning to some characters. For example:
+  - `\d` matches one digit, from 0 to 9
+  - `\D` matches character that is not a digit
+
+## Important links
+
+- [Regex Tutorial](https://regexone.com/)
+- [Regex Cheat Sheet](https://www.rexegg.com/regex-quickstart.html)
+- [Regex Testing Tool: regex101](https://regex101.com/): Select the flavor 'ECMAScript (JavaScript)'
+- Implemented using [JavaScript's RegExp implementation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions)
+- Supports [JavaScript RegExp Flags](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions#advanced_searching_with_flags), although not all of them are relevant in this context.
+
+## Known limitations
+
+Please be aware of the following limitations in Tasks' implementation of regular expression searching:
+
+- The single error message `Tasks query: cannot parse regex (description); check your leading and trailing slashes for your query` may mean any of:
+  - The opening or closing `/` is missing from the query.
+  - The regular expression is not valid, for example `description regex matches /[123/`.
+- No error when part of the pattern is lost, for example because unescaped slashes are used inside the pattern.
+  - For example, `path regex matches /a/b/c/d/` actually searches for `path regex matches /a/`.
+  - In this case, the query should be `path regex matches /a\/b\/c\/d/`.
+- Illegal flags are ignored.
+  - For example, the query `description regex matches /CASE/&` should give an error that `&` (and similar) are unrecognised flags.
+- The `tag` or `tags` instruction does not yet support regular expression searches.
+- [Lookahead and Lookbehind](https://www.regular-expressions.info/lookaround.html) searches are untested, and are presumed not to work on Apple mobile devices, or to cause serious performance problems with slow searches.
+
+## Regular expression examples
+
+Example searches:
```

Comment:
Include an example of searching for a tag, perhaps one without sub-tags. (there was a question about searching for `#t` I think)

Reply:
While you say that, including a nested tag example would likely be helpful specifically because escaping is important if matching multiple levels of the tag due to `/`

Reply:
link to the appropriate section of the regexp examples (separate file)?